client-gen/src/generator.rs

```diff
@@ -198,6 +198,7 @@ impl<'ast> Visitor<'ast> for CtorFactoryGenerator {
         self.code.push_str(
             "
             #[allow(unused)]
+            #[allow(clippy::new_without_default)]
```

Comment:
Let's add:
1. `#[derive(Default)]` to the factotry itself
2. Make it `pub struct {}Factory(())` so it can't be constructed from the outside unless when it is done either via default or new